CRM/Core/Permission/WordPress.php

```diff
@@ -58,7 +58,20 @@ public function check($str) {
       return TRUE;
     }
 
-    require_once ABSPATH . WPINC . '/pluggable.php';
+	//CRM-15629
+	// During some extern/* calls we don't bootstrap CMS hence
+	// below constants are not set. In such cases, we don't need to
+	// check permission, hence directly return TRUE
+	if ( ! defined( 'ABSPATH' ) || ! defined( 'WPINC' ) ) {
```

Comment:
If this logic is to remain present here, it seems to me that there should be a param passed to this method that allows the decision to be made whether or not to bootstrap WordPress. Otherwise, there will always be a conflict between this line and line 68 below.